src/SFA.DAS.Employer.PR.Web/Infrastructure/Services/UserRolesService.cs

```diff
@@ -0,0 +1,58 @@
+﻿using Newtonsoft.Json;
+using SFA.DAS.Employer.PR.Domain.OuterApi.Responses;
+using SFA.DAS.Employer.PR.Web.Authentication;
+using System.Security.Claims;
+
+namespace SFA.DAS.Employer.PR.Web.Infrastructure.Services;
+
+public class UserRolesService : IUserRolesService
+{
+    private readonly ILogger<UserRolesService> _logger;
+
+    public UserRolesService(ILogger<UserRolesService> logger)
+    {
+        _logger = logger;
+    }
+
+    public bool CheckUserAccountAccess(ClaimsPrincipal user, string employerAccountId, EmployerUserRole userRoleRequired)
```

Comment:
This can be extension to `ClaimsPrincipal` Also can we avoid using Newtonsoft?

Reply:
I've removed newtonsoft references, and I like the idea of moving this to claimsprincipal, so will do